File: truss-compiler/src/main/java/truss/compiler/Builder.java
package truss.compiler;

import org.apache.commons.lang.Validate;

import java.io.IOException;

public class Builder {
    private final TrussArguments arguments;

    public Builder(TrussArguments arguments) {
        Validate.notNull(arguments, "arguments");

        this.arguments = arguments;
    }

    public int build() throws IOException {
        return 0;
    }
}

File: Truss.Compiler/Builder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler {
    public class Builder {
        private readonly TrussArguments _arguments;

        public Builder(TrussArguments arguments) {
            if (arguments == null) {
                throw new ArgumentNullException("arguments");
            }

            _arguments = arguments;
        }

        public int Build() {
            return 0;
        }
    }
}
